@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1302,6 +1309,79 @@ export class TextElementsRenderer {
             const layer = textCanvas.getLayer(textElement.renderOrder || DEFAULT_TEXT_CANVAS_LAYER);
 
             const isPathLabel = textElement.path !== undefined && !textElement.isLineMarker;
+            let screenPoints: THREE.Vector2[];
+            let indexOfFirstVisibleScreenPoint = -1;
+
+            if (isPathLabel) {
+                // Get the screen points that define the label's segments and create a path with
+                // them.
+                screenPoints = [];
+                let minX = Number.MAX_SAFE_INTEGER;
+                let maxX = Number.MIN_SAFE_INTEGER;
+                let minY = Number.MAX_SAFE_INTEGER;
+                let maxY = Number.MIN_SAFE_INTEGER;
+                const tileCenterX = textElement.tileCenterX!;
+                const tileCenterY = textElement.tileCenterY!;
+                for (const pt of textElement.path!) {
+                    tempPosition.set(pt.x + tileCenterX, pt.y + tileCenterY, 0);
+                    const screenPoint = this.m_screenProjector.project(tempPosition);
+                    if (screenPoint === undefined) {
+                        continue;
+                    }
+                    screenPoints.push(screenPoint);
+
+                    if (screenPoint.x < minX) {
+                        minX = screenPoint.x;
+                    }
+                    if (screenPoint.x > maxX) {
+                        maxX = screenPoint.x;
+                    }
+                    if (screenPoint.y < minY) {
```

Comment:
I think you would save some code by using THREE.Box2 and expandByPoint

Reply:
Saves lines of code, but I would rather go for speed here.

Reply:
How is it faster? expandByPoint just calls min / max on both points, see: https://github.com/mrdoob/three.js/blob/cba14ad6baf2d00c2e2198b67a4266fe044062ce/src/math/Vector2.js#L252

If I can learn something in the process, that is also good :)

Reply:
The overhead is is in the calls to expandBy and Math.min/max, and the extra referencing of the x/y properties in the objects.